code/starter_script - Copy.R

```diff
@@ -69,40 +67,50 @@ max_process_metrics = cal_max_temp_timing(data_in = rgnc_by_seg_date,
 max_hybrid_metrics = cal_max_temp_timing(data_in = rgnc_by_seg_date, 
                                            observe_col = temp_c, predict_col = rgcn2_full_temp_c, 
                                            date_col = date)
-## calculating Nash coefficient of efficiency (CE;Nash and Sutcliffe)
+
+## Creating a funcion to calculate Nash coefficient of efficiency (Nash):
 cal_nash <- function(observe_col, predict_col, n_digits = 2){
-  nash = round(1 - (sum(observe_col - predict_col) ^ 2) / (sum(observe_col - mean(observe_col) ^2)), digits = n_digits)  
+  nash = round(1 - ((sum((observe_col - predict_col) ^ 2)) / (sum((observe_col - mean(observe_col)) ^2))), digits = n_digits)  
 }
-# nse values > 1 which mean the metric isn't calculting correctly.
-nse_metric =rgnc_by_seg_date %>%
-  group_by(seg_id_nat, lubridate::year(date)) %>%
+# calculating Nash coefficient of efficiency (CE;Nash and Sutcliffe) values: 
+nse_metric =rgnc_by_seg %>%
   summarize(nse_process = cal_nash(temp_c, sntemp_temp_c),
             nse_hybrid = cal_nash(temp_c, rgcn2_full_temp_c))
-## Calculating the exceedence metric. 
-# Using if statements.  Not working
-cal_exceedance_NOT <- function(observe_col, predict_col, temp_threshold = 25.5){
-if ({{observe_col}} >= temp_threshold & {{predict_col}} >= temp_threshold){
-    return("TRUE")
-  }else if ({{observe_col}} >= temp_threshold & {{predict_col}} < temp_threshold ){
-    return("False_negative")
+# To view the data associated with this segment we can filter through the dataset:
+View(filter(rgnc_dat_filter, seg_id_nat %in% 1549)) #filter(data, column_name %in% the certian group)
+
+## Calculating the exceedance metric. 
+# Using if statements.
+calc_exceedance <- function(observe_col, predict_col, metric, temp_threshold = 25.5){
+  observe_exceeds <- observe_col > temp_threshold
+  predict_exceeds <- predict_col > temp_threshold
+  
+  if (metric == 'prop_correct') {
```

Comment:
Now you can get rid of the `prop_correct` calculation since you've covered that with `prop_true_pos` and `prop_true_neg`